src/test/java/org/takes/tk/TkProxyTest.java

```diff
@@ -49,23 +51,273 @@
+                            "Content-Length: 134",
+                            "Transfer-Encoding: Options"
+                        ),
+                        Matchers.containsString(testText)
+                    );
+                }
+            }
+        );
+    }
+
+    /**
+     * TkProxy can work with Head HTTP method.
+     * Plus 4 random headers checked.
+     * @throws Exception If some problem inside
+     */
+    @Test
+    public void actsOnHttpHeadWithRequestHeaders() throws Exception {
+        final String testText = "HeadTest";
+        new FtRemote(new TkFixed(testText)).exec(
+            new FtRemote.Script() {
+                @Override
+                public void exec(final URI home) throws IOException {
+                    MatcherAssert.assertThat(
+                        TkProxyTest.this.actResponse(
+                            home,
+                            HttpMethod.HEAD,
+                            "TestHeader: HeadTest",
+                            "SomeHeader: HeadTestValue",
+                            "Content-Length: 135",
+                            "Transfer-Encoding: Head"
+                        ),
+                        Matchers.containsString("OK")
+                    );
+                }
+            }
+        );
+    }
+    /**
+     * TkProxy can throw IOException on invalid HTTP methods.
+     * @throws Exception If some problem inside.
+     */
+    @Test(expected = IOException.class)
+    public void actsOnInvalidHttpMethods() throws Exception {
+        new FtRemote(new TkFixed("InvalidHeaderTest")).exec(
+            new FtRemote.Script() {
+                @Override
+                public void exec(final URI home) throws IOException {
+                    TkProxyTest.this.actResponse(
+                        home, "INVALIDHTTPMETHOD"
                     );
                 }
             }
         );
     }
+
+    /**
+     * TkProxy can proxy original response headers.
+     * @throws Exception If some problem inside.
+     */
+    @Test
+    public void actsOnResponseHeaders() throws Exception {
+        final String testString = "TestRH";
+        new FtRemote(
+                new TkFixed(testString)
+        ).exec(
+            new FtRemote.Script() {
+                @Override
+                public void exec(final URI home) throws IOException {
+                    MatcherAssert.assertThat(
+                        new RsPrint(
+                            new TkProxy(
+                                proxy(home)
+                            ).act(new RqFake())
+                        ).print(),
+                            Matchers.allOf(
+                                        Matchers.containsString(testString),
+                                        Matchers.containsString(
+                                                "X-Takes-TkProxy:"
+                                        )
+                                )
+                        );
```

Comment:
@aschworer this method also is rather messy, in terms of indentation. please, use exactly 4 spaces, always